Marlin/src/module/motion.cpp

```diff
@@ -1509,8 +1509,8 @@ void homeaxis(const AxisEnum axis) {
         }
         else {
           // In other modes, T0 can move from X_MIN_POS to X_MAX_POS
-          soft_endstop_min[axis] = base_min_pos(axis);
-          soft_endstop_max[axis] = base_max_pos(axis);
+          soft_endstop_min[axis] = X1_MIN_POS;
+          soft_endstop_max[axis] = X1_MAX_POS;
```

Comment:
Ill take a look at what sets those values. If those are not updated per x1 and x2, crash condition still exists.

Reply:
The `base_...` macros are just aliases to get items from a `const` array. If there's anything weird going on, then to my mind it tends to implicate the use of `PROGMEM` in the definition of `XYZ_CONSTS`.

Reply:
The values I placed in are config file values from the dual x carriage segment that limit the travel of the specific tool head, so they may not match. In this case, the base_max_pos for x axis is 442, but the last 40mm is the park position of X2 so x1_max_pos is 400. This prevents it from trying to go through X2 while parking, or allowing you to jog x1 into x2.

Reply:
The max position for X1 should already be configured so that it can only go as far as being next to X2. But, since the X1_MIN_POS and X1_MAX_POS can be overridden, your change here is fine.